compiler/luci/service/src/Nodes/CircleReshape.test.cpp

```diff
@@ -197,3 +250,79 @@ TEST(ShapeRuleTest, reshape_by_newShape)
   ASSERT_EQ(2, output_shape.dim(0).value());
   ASSERT_EQ(12, output_shape.dim(1).value());
 }
+
+TEST(ShapeRuleTest, reshape_by_newShape_dynamic)
+{
+  auto g = loco::make_graph();
+  auto node_reshape = g->nodes()->create<luci::CircleReshape>();
+  auto tensor_input = g->nodes()->create<luci::CircleInput>();
+  auto shape_dummy = g->nodes()->create<luci::CircleOutputDummy>();
+
+  tensor_input->dtype(loco::DataType::S32);
+  tensor_input->shape({2, 3, 4});
+  tensor_input->shape_status(luci::ShapeStatus::VALID);
+
+  shape_dummy->dtype(loco::DataType::S32);
+  shape_dummy->shape({});
+  shape_dummy->shape_status(luci::ShapeStatus::VALID);
+
+  node_reshape->tensor(tensor_input);
+  node_reshape->shape(shape_dummy);
+
+  // reshape to {-1, 12}
+  node_reshape->newShape()->rank(2);
+  node_reshape->newShape()->dim(0) = -1;
+  node_reshape->newShape()->dim(1) = 12;
+
+  loco::TensorShape output_shape;
+  luci::sinf::Rule shape_inf_rule;
+
+  ASSERT_TRUE(shape_inf_rule.infer(node_reshape, output_shape));
+
+  ASSERT_EQ(2, output_shape.rank());
+  ASSERT_FALSE(output_shape.dim(0).known());
```

Comment:
Why is this unknown? I guess that Reshape (2,3,4) into (-1, 12) should have returned (2, 12)?

Reply:
Hmm, good point. My intention here was to set as dynamic, but you are right that -1 have also special meaning.
I debug it more deeply and IHMO [is_static_shape = false](https://github.com/Samsung/ONE/blob/e5e151d547616f1ad3be6ff13f86b53990b7c157/compiler/luci/service/src/Nodes/CircleReshape.cpp#L134) is not needed here. As a result [-1 handling here](https://github.com/Samsung/ONE/blob/e5e151d547616f1ad3be6ff13f86b53990b7c157/compiler/luci/service/src/Nodes/CircleReshape.cpp#L184) is not called.

But it seems to be sth separately to the main goal of my change (especially if even now it's not very clear😉)

To avoid confusions I've changed the test data.